sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ProactiveContainerInitConfig.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.util.List;
+
+/**
+ * Encapsulates the list of container identities and no. of proactive connection regions.
+ * */
+public final class ProactiveContainerInitConfig {
+    private final List<CosmosContainerIdentity> cosmosContainerIdentities;
+    private final int numProactiveConnectionRegions;
+
+    ProactiveContainerInitConfig(List<CosmosContainerIdentity> cosmosContainerIdentities, int numProactiveConnectionRegions) {
+        this.cosmosContainerIdentities = cosmosContainerIdentities;
+        this.numProactiveConnectionRegions = numProactiveConnectionRegions;
+    }
+
+    /**
+     * Gets the list of container identities
+     * */
+    public List<CosmosContainerIdentity> getCosmosContainerIdentities() {
+        return cosmosContainerIdentities;
+    }
+
+    /**
+     * Gets the no. of proactive connection regions
```

Comment:
Please go into more details here - like what is a proactive connection region - how is the special case in single-master with write region outside of number of preferred regions - like numberOfPreferredRegions is 1, Customer has preferred regions "East US" and "West US" - but Write region is "West US". We wouldn't create connections then - but some customers might the write region to always be included. We should call  this out here.